packages/material-ui-docs/src/MarkdownElement/MarkdownElement.js

```diff
@@ -19,11 +19,15 @@ marked.Lexer.prototype.lex = function lex(src) {
 const renderer = new marked.Renderer();
 
 export function textToHash(text) {
-  return text
-    .toLowerCase()
-    .replace(/=&gt;|&lt;| \/&gt;|<code>|<\/code>/g, '')
-    .replace(/\W+/g, '-')
-    .replace(/-$/g, '');
+  return encodeURI(
+    text
+      .toLowerCase()
+      .replace(/=&gt;|&lt;| \/&gt;|<code>|<\/code>/g, '')
+      .replace(/\s+/g, '-')
+      // eslint-disable-next-line no-useless-escape
+      .replace(/[!@#\$%\^&\*\(\)=_\+\[\]{}`~;:'"\|,\.<>\/\?]+/g, '')
+      .replace(/-$/g, ''),
+  );
 }
```

Comment:
This is the "meat and potatoes" of this PR. It enables fragment ids to work for UTF8 headings.

Reply:
(But, "no-useless-escape" seems fubar.)